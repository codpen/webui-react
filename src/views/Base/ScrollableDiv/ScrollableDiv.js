import React from "react";

function ScrollableDiv({height, children}) {
    return (<div style={{overflow: "auto", height: height}}>
        {children}
    </div>)
}

export default ScrollableDiv;
