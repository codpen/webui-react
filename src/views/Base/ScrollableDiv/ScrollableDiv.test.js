import React from "react";
import {shallow} from "enzyme";
import {testStore} from "../../../../Utils";
import toJson from "enzyme-to-json";
import ScrollableDiv from "./ScrollableDiv";

const setUp = (intialState = {}, props = {}) => {
    const store = testStore(intialState);
    const component = shallow(<ScrollableDiv {...props} store={store}/>);
    return component;
};


describe('Running Jobs', function () {


    describe('renders', function () {
        let wrapper;
        beforeEach(() => {
            const initialState = {};

            const props = {
                height: '700px',
                children: <div/>
            };
            wrapper = setUp(initialState, props)
        });

        it('should render without crashing', function () {
            expect(wrapper).toHaveLength(1)
        });

        it('should match snapshot', function () {
            expect(toJson(wrapper)).toMatchSnapshot()
        });

    });


});