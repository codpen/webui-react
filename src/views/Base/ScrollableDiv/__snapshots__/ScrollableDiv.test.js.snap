// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Running Jobs renders should match snapshot 1`] = `
<div
  style={
    Object {
      "height": "700px",
      "overflow": "auto",
    }
  }
>
  <div />
</div>
`;
